validation/views.py

```diff
@@ -707,6 +734,59 @@ def save_wrong_word(request, recording_id):
     return response
 
 
+@login_required()
+def record_audio(request):
+    if request.method == "POST":
+        form = RecordNewPhrase(request.POST, request.FILES)
+        translation = request.POST.get("translation")
+        transcription = request.POST.get("transcription")
+        audio_data = request.FILES["audio_data"]
+
+        phrase = Phrase.objects.filter(transcription=transcription).first()
+        if not phrase:
+            phrase = Phrase(
+                translation=translation,
+                field_transcription=transcription,
+                transcription=transcription,
+                kind=Phrase.SENTENCE if " " in transcription else Phrase.WORD,
+                status=Phrase.USER,
+                date=datetime.datetime.now(),
+                modifier=request.user,
+            )
+            phrase.save()
+
+        speaker = Speaker.objects.filter(user=request.user).first()
+        if not speaker:
+            speaker = Speaker(
+                full_name=request.user.first_name + " " + request.user.last_name,
+                code=request.user.first_name[:3].upper(),
+            )
+            speaker.save()
+
+        _id = create_new_rec_id(phrase, speaker)
+        audio_data.name = _id + ".wav"
+        rec = Recording(
+            id=_id,
+            compressed_audio=audio_data,
+            speaker=speaker,
+            phrase=phrase,
+            timestamp=0,
```

Comment:
If this is a float, could you store the actual current time, `time.time()` as a timestamp? The metadata of when the user-recording was uploaded will be handy in the future.

Actually I’d recommend:
  - Use a filename that is not a hash, e.g., `u${user.id}-${timestamp_with_millis}.wav`
  - Next to `foo.wav`, write out `foo.json` with all the original metadata for posterity: user-supplied transcript / translation, user id + login + full name, and timestamp

That way it’ll be much easier to potentially reuse these files for other things down the road. Basically anything you wished had just been in a JSON file next to every original audio file in the recording sessions, so you wouldn’t have had to parse it out of filenames or something: put all that in here from the start.

Reply:
Ooh okay, two things:
1. `timestamp` is a bit misleading: it's the time, in milliseconds, when this utterance occurs in the larger .wav file, not the time of day it was recorded
2. Yes. Yes please. This is my new priority with this project, second only to seeing if I can actually save these files on the server side.